### DISCUSSION TITLE
Downloading blastdb nr database can't --decompress

### DISCUSSION THREAD

**Discussion by kuheng3** [biostars/biostar-handbook-issues#191] [General]
Hi, 
I have tried to download nr database using "update_blastdb.pl nr --decompress". Downloading the database is fine but when it start decompressing files. I keep getting:

kuheng@Ku ~/blast/db/nr
$ update_blastdb.pl nr --decompress
Connected to NCBI
Downloading nr (55 volumes) ...
Decompressing nr.00.tar.gz ...Read error on tarfile (missing data) 'nr.pdb' at offset unknown at /Users/kuheng/miniconda3/envs/bioinfo/bin/update_blastdb.pl line 496.
Invalid header block at offset unknown at /Users/kuheng/miniconda3/envs/bioinfo/bin/update_blastdb.pl line 496.
Invalid header block at offset unknown at /Users/kuheng/miniconda3/envs/bioinfo/bin/update_blastdb.pl line 496.
Invalid header block at offset unknown at /Users/kuheng/miniconda3/envs/bioinfo/bin/update_blastdb.pl line 496.
Is there any solution to fix the update_blastdb.pl? 
Thank you

**Comment by ialbert** [comment]
Can you try adding an additional parameter:

    --source ncbi 

the source should be automatically detected but sometimes it is not set properly

**Comment by kuheng3** [comment]
Hi Dr. Ialbert, 
I tried update_blastdb.pl nr --source ncbi --decompress. Here is the result I got:
kuheng@Ku ~/blast/db/nr
$ update_blastdb.pl nr --source ncbi --decompress
Connected to NCBI
Downloading nr (55 volumes) ...
Decompressing nr.00.tar.gz ...Read error on tarfile (missing data) 'nr.pdb' at offset unknown at /Users/kuheng/miniconda3/envs/bioinfo/bin/update_blastdb.pl line 496.
Invalid header block at offset unknown at /Users/kuheng/miniconda3/envs/bioinfo/bin/update_blastdb.pl line 496.

**Comment by ialbert** [comment]
I have now tried it myself,  indeed it raises an error for me as well. 

Goes to show that not even NCBI is capable of publishing data/tools as simple as one that just downloads data. What hope is there for regular scientists?

A possible workaround is to manually decompress the file that is already there:

    tar zxvf nr.00.tar.gz 

then reissue the original command again, 

    update_blastdb.pl nr --source ncbi --decompress

it should continue on with downloading and decompressing the next file, I did not check if those work or not.

For a deeper understanding of what is going on, the tool automates the downloads from this location:

https://ftp.ncbi.nlm.nih.gov/blast/db/

to do the same from command line you would do:

    wget https://ftp.ncbi.nlm.nih.gov/blast/db/nr.00.tar.gz
    
followed by:

     tar zxvf nr.00.tar.gz 
     
If the original command does not work, now you need to automate that for all 54 files :-/

**Comment by kuheng3** [comment]
Hi Dr. Ialbert,
Thank you so  much, I will try the methods you mentioned.  I wasn't sure if is my computer setting issue or the perl script. If the "tar zxvf nr.00.tar.gz" + "update_blastdb.pl nr --source ncbi --decompress" didn't work. I will use a script to unzip all nr.tar.gz files.
